frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,107 @@
+import localforage from 'localforage'
+import {toHash, makeEntry, makeEvent, makeLog} from '../../../shared/functions'
+import request from 'superagent'
+import {EVENTS} from '../../../shared/events'
+import {Group} from '../../../shared/types'
+const {CREATION} = EVENTS
+var store
+// =======================
+// Helpers
+// =======================
+let _db = new Map()
+function getDB (hash: string) {
+  if (_db.has(hash)) {
+    return _db.get(hash)
+  }
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  _db.set(hash, db)
+  return db
+}
+// =======================
+// Exports
+// =======================
+// getItemFromLog retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export default {
+  async getItemFromLog (hash: string) {
+    if (!store.state.loggedInUser) {
+      return
+    }
+    let groupId = store.state.session.currentGroup.groupId
+    let db = getDB(groupId)
+    let entry = await db.getItem(hash)
+    return entry
+  },
+  // addItemToLog appends to the log
+  // returns the current state of the log
+  async addItemToLog (value: any) {
+    if (!store.state.loggedInUser) {
+      return
+    }
+    let groupId = store.state.session.currentGroup.groupId
+    let db = getDB(groupId)
+    let currentLogPosition = await db.getItem('currentLogPosition')
+    let entry = makeEntry(value, currentLogPosition)
+    let hash = toHash(entry)
+    await request.post(`${process.env.API_URL}/event/${groupId}`)
+        .send({ hash, entry })
+    await db.setItem(hash, entry)
+    await db.setItem('currentLogPosition', hash)
+    store.commit('updateCurrentLogPosition', hash)
```

Comment:
- These three lines should be moved to a store *action*, and instead of calling `commit` here call `dispatch` to dispatch that action.
- Alternative: this function is itself made an action and is mixed in to the store actions using object-spread operator.